### PR TITLE
fix: fix broken Sparse Float Vector raw data mmap

### PR DESCRIPTION
**Pull request by zhengbuqian**
issue: https://github.com/milvus-io/milvus/issues/36182

* improved `Column.h` to make the code much more readable and maintainable, and added detailed comments.
* fixed an issue where `ArrayColumn::NumRows()` always returns 0 when the mmap backing storage is a file.
* removed unused `ColumnBase` constructors and unnecessary members so we don't get confused.
* Updated `test_chunk_cache.cpp` to make the tests parameterized: to test both mmap enabled and disabled. Added sparse field in the test to add coverage.
* re-enabled test `Sealed::GetSparseVectorFromChunkCache`. 
    * But 2 other disabled tests `Sealed::WarmupChunkCache` and `Sealed::GetVectorFromChunkCache` remain disabled, there seems to be errors. @bigsheeper PTAL.

### CODE REVIEW

File: internal/core/src/mmap/Column.h [hunk 1]
```diff
@@ -210,36 +231,22 @@ class ColumnBase {
...
             if (mapping_type_ != MappingType::MAP_WITH_MANAGER) {
-                size_t mapped_size = data_cap_size_ + padding_;
                 if (munmap(data_, mapped_size)) {
                     AssertInfo(true,
                                "failed to unmap variable field, err={}",
                                strerror(errno));
                 }
             }
-            UpdateMetricWhenMunmap(data_cap_size_ + padding_);
+            UpdateMetricWhenMunmap(data_cap_size_ + mapped_size);
```
Comment (congqixia) [inline internal/core/src/mmap/Column.h:242]: maybe
```suggestion
            UpdateMetricWhenMunmap(mapped_size);
```

**Comment by zhengbuqian** [comment]
rerun go-sdk

**Comment by zhengbuqian** [comment]
/unhold

pushed a new commit to correctly set SparseColumn dim_ at MAP_WITH_FILE mode. https://github.com/milvus-io/milvus/pull/36183/commits/fa91d85c5f4bf49b5ab49eb1e5bdb2b751de63c5

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**